python/cudf/cudf/utils/ioutils.py

```diff
@@ -2099,3 +2108,96 @@ def _read_byte_ranges(
 
     for worker in workers:
         worker.join()
+
+
+def _get_remote_bytes_all(
+    remote_paths, fs, *, blocksize=_BYTES_PER_THREAD_DEFAULT
+):
+    # TODO: Experiment with a heuristic to avoid the fs.sizes
+    # call when we are reading many files at once (the latency
+    # of collecting the file sizes is unnecessary in this case)
+    if max(sizes := fs.sizes(remote_paths)) <= blocksize:
+        # Don't bother breaking up individual files
+        return fs.cat_ranges(remote_paths, None, None)
+    else:
+        # Construct list of paths, starts, and ends
+        paths, starts, ends = zip(
+            *[
+                (r, j, min(j + blocksize, s))
+                for r, s in zip(remote_paths, sizes)
+                for j in range(0, s, blocksize)
+            ]
```

Comment:
*nit*: no need to cons the list before the splat.
```suggestion
            *(
                (r, j, min(j + blocksize, s))
                for r, s in zip(remote_paths, sizes)
                for j in range(0, s, blocksize)
            )
```

Reply:
Good call. Note that `cat_ranges` actually needs `paths`, `starts`, and `ends` to each be a list. Therefore, this probably needs to be something like:

```python
        paths, starts, ends = map(
            list,
            zip(
                *(
                    (r, j, min(j + blocksize, s))
                    for r, s in zip(remote_paths, sizes)
                    for j in range(0, s, blocksize)
                )
            ),
        )
```

I'll need to tweak test coverage to hit this case.